### PR TITLE
Platform/jimfuqian/11928 refactor code to remove old dependencies resurrect

### LINKED ISSUE #11928: Re-surrect-PR: Refactor code to remove old dependencies
### User Story
As a ReportStream engineer, I want to keep our dependencies current so that use of unmaintained code is minimized.
Note, this round of work is to re-surrect the PR which was reverted from last de[ployment.

### Description/Use Case
A number of dependencies (see Dev Notes below for a list) appear to be no longer maintained or public. We should remove these dependencies and refactor any code that uses them to use alternatives.

### Dev Notes
Mo supplied these notes:

- `com.github.kittinunf.fuel:fuel-jackson:2.3.1` - we use (used?) fuel to invoke web services and use Jackson to serialize data to and from JSON/Yaml. This library helped bridge those two. We could replace fuel with Ktor and yoink this one
- `com.github.KennethWussmann:mock-fuel:1.3.0` - ditto above. We needed this to mock our Fuel calls. I don't think this is maintained anymore 
- `com.jcraft:jsch:0.1.55` - older SSH/SFTP library that's also no longer maintained. This one can probably get pulled entirely (and should be) it's almost certainly not being used actively. Prune it.

Additionally, `mock-fuel` appears to be the only remaining use of `jitpack`, which should also be removed (see #11938).

### Acceptance Criteria
- [x] Dependency `com.github.kittinunf.fuel:fuel-jackson:2.3.1` is removed or an explanation for its continued use is provided.
- [x] Dependency `com.github.KennethWussmann:mock-fuel:1.3.0` is removed or an explanation for its continued use is provided.
- [x] Dependency `com.jcraft:jsch:0.1.55` is removed or an explanation for its continued use is provided.
- [x] Dependency `jitpack.io` is removed or an explanation for its continued use is provided.

### PR DESCRIPTION
**Pull request by thetaurean**
This PR:

Brings back a past branch that was unable to past server2server auth tests. The problem was an unregistered callback for handling a refresh token event. Original PR text below for clarity / information.

Removes unmaintained libraries around Fuel http: https://github.com/kittinunf/fuel and fuel mocks used with tests.
This removal improves report stream code base security posture.

the replacement is ktor http: https://github.com/ktorio/ktor

Test Steps:
1. Run full test suite to include server2server auth tests. 
  ```
   ./prime test --env staging --run=server2serverauth --key $AZURE_key
   ```

## Changes
- Changed gradle.build.kts to remove Fuel and all related dependencies including fuel mocks and added any additional dependencies required by ktor http
- Changed main code to use ktor http in place where Fuel http used
- Created HttpClientUtils.kt where GET, POST, PUT, HEAD, DELETE, SubmitForm are wrapped and exposed as handy utilities functions to the report stream main code, ktor http client "Content Negotiation" is configured with the ktor http client to promote clean code with regard to response body to serializable classes.
- Refactor tests accordingly
- Note: there were transports classes already using ktor http instead of Fuel before this PR, those are not touched, but tickets are logged for future consolidation
- Note, there are other http client used in the code base, e.g. azure http, those are not touched since there is no concern about the log term maintain etc.

### Testing
- [X] Tested locally?
- [X] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [X] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [X] Added tests? some old tests are replaced with tests using ktor mock engine

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #11928 

## To Be Done
- Investigate other uses of HttpClient and make sure refresh token is adequately handled #issue

### CODE REVIEW

File: prime-router/build.gradle.kts
Comment (jalbinson) [inline prime-router/build.gradle.kts:452]: Is this task no longer necessary?

File: prime-router/src/main/kotlin/azure/EmailEngineFunction.kt
Comment (jalbinson) [inline prime-router/src/main/kotlin/azure/EmailEngineFunction.kt:332]: move this to 

```kotlin
logger.warning("Error in fetching emails, exception", ex)
```

So it triggers errors and we catch that something is amiss.

File: prime-router/build.gradle.kts
Comment (jalbinson) [inline prime-router/build.gradle.kts:938]: Why did we add these libraries?

Generally anything scoped to just tests is not a big deal but it adds more dependencies to keep up to date which we're specifically trying to reduce in this ticket.

File: prime-router/src/test/kotlin/cli/LookupTableCommandsTest.kt
Comment (jalbinson) [inline prime-router/src/test/kotlin/cli/LookupTableCommandsTest.kt:178]: Its better to assert the exception type than the message within

File: prime-router/src/main/kotlin/common/HttpClientUtils.kt
Comment (jalbinson) [inline prime-router/src/main/kotlin/common/HttpClientUtils.kt:499]: We have a single threaded model at the moment but in general it would be better to make these functions async.

```kotlin
     private suspend fun invoke() {
     // runBlocking no longer needed and will be handled at the caller (or not if we're doing things the right way)
```

File: prime-router/src/main/kotlin/common/HttpClientUtils.kt
Comment (jalbinson) [inline prime-router/src/main/kotlin/common/HttpClientUtils.kt:557]: Since this does a ton under the hood it would probably be easier to just nix it and add the header by hand.

```kotlin
bearerTokens?.accessToken?.let {
    headers {
        append("Authorization", "Bearer $it")
    }
}
```

File: prime-router/build.gradle.kts
Comment (thetaurean) [reply prime-router/build.gradle.kts:452]: Not sure why it was removed. I recall discussion around merging the two -- testEnd2End and testEnd2EndUP -- but I haven't found anything concrete there so I've added it back in for now.

File: prime-router/build.gradle.kts
Comment (thetaurean) [reply prime-router/build.gradle.kts:938]: Nice catch! These were not necessary. I've reached out to Jim to find out why they were added but for now I've removed them.

File: prime-router/src/test/kotlin/cli/LookupTableCommandsTest.kt
Comment (thetaurean) [reply prime-router/src/test/kotlin/cli/LookupTableCommandsTest.kt:178]: There's an assertion above on the exception type. This assertion is further verifying the message content which seems appropriate in the context. We should make that exception unique but feels out of scope for now.

File: prime-router/src/main/kotlin/common/HttpClientUtils.kt
Comment (thetaurean) [reply prime-router/src/main/kotlin/common/HttpClientUtils.kt:499]: Changing this litters the code with `runBlocking {}` calls. I suggest we cross this bridge when we get there, it seems be simpler to transition to multithread as currently written.

File: prime-router/src/main/kotlin/azure/EmailEngineFunction.kt
Comment (thetaurean) [reply prime-router/src/main/kotlin/azure/EmailEngineFunction.kt:332]: I wasn't able to chain that exception to `logger.warn`, maybe I'm missing something?

File: prime-router/src/main/kotlin/common/HttpClientUtils.kt
Comment (jalbinson) [reply prime-router/src/main/kotlin/common/HttpClientUtils.kt:498]: I agree. It would be nice to at some point to correctly handle coroutines but this is not the ticket to do it in. I was just bringing this design decision up.

File: prime-router/src/main/kotlin/common/HttpClientUtils.kt
Comment (jalbinson) [reply prime-router/src/main/kotlin/common/HttpClientUtils.kt]: I think thats a premature optimization though. We currently don't use refresh tokens and it caused us an issue. I think these few lines are what we actually want to do at the moment and if we do want to use refresh tokens in the future it will be easy to find where the auth header is currently being appended.

Comment (thetaurean) [reply prime-router/src/main/kotlin/common/HttpClientUtils.kt]: Yeah you're right. Copacetic :).

**Review by jalbinson — APPROVED** [review]
Looks good. I'd recommend getting another pair of eyes on this before you merge though since the changes are so far reaching.

File: prime-router/src/main/kotlin/cli/LookupTableCommands.kt
Comment (arnejduranovic) [inline prime-router/src/main/kotlin/cli/LookupTableCommands.kt:55]: I don't understand why we need this now?

File: prime-router/src/main/kotlin/transport/GAENTransport.kt
Comment (arnejduranovic) [inline prime-router/src/main/kotlin/transport/GAENTransport.kt:44]: I don't understand this change. httpClient is only used once below and it is ignore and overwritten?

File: prime-router/src/main/kotlin/credentials/HashicorpVaultCredentialService.kt
Comment (arnejduranovic) [inline prime-router/src/main/kotlin/credentials/HashicorpVaultCredentialService.kt:27]: I don't understand this change. httpClient is only used once below and it is ignore and overwritten?

**Review by arnejduranovic — APPROVED** [review]

**Pull request merged**
